test/grouping.jl

```diff
@@ -291,4 +287,19 @@ end
 	@test @select(g, :a, t = :c).a ≅ d.a
 end
 
+@testset "@where with a grouped data frame" begin
+    df = DataFrame(
+        g = [1, 1, 1, 2, 2],
+        i = 1:5,
+        t = ["a", "b", "c", "c", "e"],
+        y = [:v, :w, :x, :y, :z],
+        c = [:g, :quote, :body, :transform, missing]
+    )
+
+    gd = groupby(df, :g)
+
+    @test @where(gd, :i .== first(:i)).t == ["a", "c"]
+    @test @where(gd, cols(:i) .> mean(cols(:i)), :t .== "c").t == ["c"]
+    @test @where(gd, :c == :g).i == Int[]
```

Comment:
As usual I'd recommend testing equality against a full data frame, to ensure everything is correct and not just one column.